File: IP-Address/Resolve-IP.js
const dns = require("dns");

process.stdout.write("Enter a URL: ");

process.stdin.once("data", (data) => {
  let url = data.toString().trim();

  url = url.replace(/^https?:\/\//, "");
  url = url.replace(/^www\./, "");
  url = url.split("/")[0];

  dns.lookup(url, (err, address) => {
    if (err) {
      console.log("Could not resolve the domain");
    } else {
      console.log(`IP Address of ${url}: ${address}`);
    }
    process.exit();
  });
});
// How to run: node Resolve-IP.js

File: IP-Address/Resolve-IP.py
import socket

def clean_url(url):
    url = url.strip()
    if url.startswith("http://"):
        url = url.replace("http://", "")
    elif url.startswith("https://"):
        url = url.replace("https://", "")
    if url.startswith("www."):
        url = url.replace("www.", "")
    return url.split("/")[0]

url = input("Enter a URL: ")
domain = clean_url(url)

try:
    ip = socket.gethostbyname(domain)
    print(f"IP Address of {domain}: {ip}")
except:
    print("Could not resolve the domain")
# How to run: python Resolve-IP.py